### PR TITLE
add "Delete all cookies" to privacy settings

### PR DESCRIPTION
**Pull request by balcy**
implements basic feature request of https://github.com/ubports/morph-browser/issues/163

### CODE REVIEW

**Review by UniversalSuperBox — CHANGES_REQUESTED** [review]
I think that [QWebEngineCookieStore::deleteAllCookies()](https://doc.qt.io/archives/qt-5.11/qwebenginecookiestore.html#deleteAllCookies) would be a much better option to use here. [QQuickWebEngineProfile::cookieStore()](https://doc.qt.io/archives/qt-5.11/qquickwebengineprofile.html#cookieStore) or [QWebEngineProfile::cookieStore()](https://doc.qt.io/archives/qt-5.11/qwebengineprofile.html#cookieStore) can be used to access the cookie store.

**Comment by balcy** [comment]
hmm strange it worked with clickable... have to look into it

**Comment by balcy** [comment]
or it built successfully now. The reason for the rebase was, that I wanted to bring in the new "SharedIncognitoContext" from https://github.com/ubports/morph-browser/pull/234/commits/07572e028e575892ff40a7d285b7214fbfd3500e.
the cookies from the standard profile, and the incognito profile are both deleted.

File: src/app/webcontainer/WebappSettingsPage.qml [hunk 1]
```diff
@@ -202,6 +203,17 @@ FocusScope {
                                 dialog.confirmed.connect(clearCache);
                             }
                         }
+
+                        ListItem {
+                            objectName: "privacy.clearAllCookies"
+                            ListItemLayout {
+                                title.text: i18n.tr("Clear all cookies")
```
Comment (UniversalSuperBox) [inline src/app/webcontainer/WebappSettingsPage.qml:210]: This option is sentence cased but the two above it are Title Cased. Shall we pick one or the other?

**Review by UniversalSuperBox — CHANGES_REQUESTED** [review]

File: src/app/webcontainer/WebappSettingsPage.qml (hunk 1 above)
Comment (balcy) [reply src/app/webcontainer/WebappSettingsPage.qml:210]: ~hmm, is "Clear All Cookies" ok ?~

Comment (balcy) [reply src/app/webcontainer/WebappSettingsPage.qml:210]: On the other hand: the first 4 items are not Title Cased (Only allow ..., Domain blacklist/whitelist, Domain specific settings, Camera & microphone)
maybe it is easier to make it "Clear browsing history" and "Clear cache"
It is used in the question for the user, too.
So I would vote for this.

**Comment by balcy** [comment]
hmm the build problem is strange (adapting the Jenkinsfile seemed to have worked on branch https://github.com/ubports/morph-browser/pull/234

I think we might be able to merge it to xenial like that one: 
https://github.com/ubports/morph-browser/pull/256

**Comment by UniversalSuperBox** [comment]
@balcy, it appears that commits from the xenial branch have appeared behind your commits in this branch. Could you please re-create this PR or push the branch so that only your changes on top of the xenial branch are here?

**Pull request closed without merging**